chrome/content/zotero/preferences/preferences_sync.js

```diff
@@ -24,19 +24,125 @@
-	init: function () {
+	init: Zotero.Promise.coroutine(function* () {
 		this.updateStorageSettings(null, null, true);
-		
-		document.getElementById('sync-api-key').value = Zotero.Sync.Data.Local.getAPIKey();
-		
+
+		var username = Zotero.Prefs.get('sync.server.username');
+		var apiKey = Zotero.Sync.Data.Local.getAPIKey();
+		if (apiKey) {
+			this.displayFields(username);
+			try {
+				var keyInfo = yield Zotero.Sync.Runner.checkAccess(Zotero.Sync.Runner.getAPIClient({apiKey}));
+				Zotero.Prefs.set('sync.server.username', keyInfo.value);
+				this.displayFields(keyInfo.username);
+			} catch (e) {
+				// API key wrong/invalid
+				Zotero.alert(
+						window,
+						Zotero.getString('general.error'),
+						Zotero.getString('sync.error.apiKeyInvalid')
+				);
+				this.unlinkAccount(false);
+			}
+		}
+
+
 		// TEMP: Disabled
 		//var pass = Zotero.Sync.Storage.WebDAV.password;
 		//if (pass) {
 		//	document.getElementById('storage-password').value = pass;
 		//}
+	}),
+
+	displayFields: function (username) {
+		document.getElementById('sync-auth-form').hidden = !!username;
+		document.getElementById('sync-authorized').hidden = !username;
+		document.getElementById('sync-reset-tab').disabled = !username;
+		document.getElementById('sync-username').value = username;
+		document.getElementById('sync-password').value = '';
+
+		var img = document.getElementById('sync-status-indicator');
+		img.removeAttribute('verified');
+		img.removeAttribute('animated');
 	},
+
+
+	credentialsKeyPress: function (event) {
+		var username = document.getElementById('sync-username-textbox');
+		username.value = username.value.trim();
+		var password = document.getElementById('sync-password');
+
+		var syncAuthButton = document.getElementById('sync-auth-button');
+
+		syncAuthButton.setAttribute('disabled', 'true');
+		if (username.value.length && password.value.length) {
+			syncAuthButton.setAttribute('disabled', 'false');
+		}
+
+		if (event.keyCode == 13) {
+			event.target.blur();
+			Zotero_Preferences.Sync.setUpSync(event);
+		}
+	},
+
+
+	setUpSync: Zotero.Promise.coroutine(function* (event) {
+		var username = document.getElementById('sync-username-textbox').value;
+		var password = document.getElementById('sync-password').value;
+
+		if (username.length <= 0 || password.length <= 0) {
+			this.updateSyncIndicator(0);
+			return;
+		}
+
+		// Try to acquire API key with current credentials
+		this.updateSyncIndicator(1);
+		var apiKey = yield Zotero.Sync.Runner.getAPIKeyFromCredentials(username, password);
+		if (apiKey) {
+			Zotero.Prefs.set('sync.server.username', username);
+			this.displayFields(username);
+		}
+		else {
+			this.updateSyncIndicator(0);
+			Zotero.alert(window,
+					Zotero.getString('general.error'),
+					Zotero.getString('sync.error.invalidLogin')
+			);
+		}
+	}),
+
+	/**
+	 * Updates the auth indicator icon, depending on status
+	 * @param {int} status 
+	 */
+	updateSyncIndicator: function (status) {
+		const animated=1;
```

Comment:
Can just take a string (e.g., 'animated') as a parameter, to make calling code clearer.